data/scripts/fetch_wikipedia.py

```
@@ -0,0 +1,108 @@
+    out_dir.mkdir(parents=True, exist_ok=True)
+    files = _parquet_urls(config, split, limit)
+    if not files:
+        raise ValueError(f"no parquet files found for {config}/{split}")
+    print(f"Found {len(files)} parquet shard(s); extracting text column ...")
+    written: list[str] = []
+    for i, entry in enumerate(files, 1):
+        url = entry["url"]
+        shard = entry["url"].rsplit("/", 1)[-1].rsplit(".", 1)[0]
+        print(f"[{i}/{len(files)}] {shard} ({entry.get('size', 0) // 1048576} MB) ...", flush=True)
+        tmp = Path("/tmp") / f"wiki-{shard}.parquet"
+        urllib.request.urlretrieve(url, tmp)
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)